tests/xharness/AppRunner.cs

```diff
@@ -25,15 +24,15 @@ class AppRunner
 		readonly ICrashSnapshotReporterFactory snapshotReporterFactory;
 		readonly ICaptureLogFactory captureLogFactory;
 		readonly IDeviceLogCapturerFactory deviceLogCapturerFactory;
-		readonly IResultParser resultParser;
+		readonly ITestReporterFactory testReporterFactory;
 
 		readonly RunMode runMode;
```

Comment:
I think `runMode` should stay private. But if you really need it public, remove this one and leave the public one only